### PR TITLE
Remove mongo-oplog dependency

### PR DESCRIPTION
**Pull request by will123195**
- this will be 2.0.0
- removes dependency on `mongo-oplog`
- adds support for async event handlers with stream backpressure

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -346,19 +352,20 @@ class MongoOplogReader extends EventEmitter {
...
-        oplog.on('op', data => {
-          if (data.fromMigrate) return; // ignore shard balancing ops
-          if (data.op === opCode.NOOP) return; // ignore informational no-operation
-          this.processOp(data, replSetName, memberName);
-          this.emit('shard-op', { data, replSetName, memberName });
-        });
-        oplog.on('end', () => {
+        const stream = createOplogStream(db, opts);
+        this.oplogs[connStr] = stream;
+        stream.pipe(through2.obj((data, enc, done) => {
```
Comment (will123195) [inline src/index.js]: I will change this to process multiple oplog docs concurrently per worker.

File: src/createOplogStream.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  if (startAt && typeof startAt === 'number') return Timestamp(0, startAt);
+  if (startAt) return startAt;
+  return Timestamp(0, Date.now() / 1000 | 0);
+}
+
+/**
+ * Create a mongo oplog stream
+ * @param db
+ * @param options
+ *   - startAt {Integer|} Oplog Timestamp to being stream, otherwise stream from now
```
Comment (wwbenomathew) [inline src/createOplogStream.js]: `being` or `beginning of` ?

File: README.md [hunk 3]
```diff
@@ -32,9 +37,16 @@ const connectionStrings = [
   'mongodb://shard1-secondary1/local'
 ];
 
-const reader = new MongoOplogReader({ redisClient });
+const reader = new MongoOplogReader({ 
+  redisClient,
+  workersPerOplog: 1
```
Comment (wwbenomathew) [inline README.md]: Would `1` invalidate https://github.com/WW-Digital/mongo-oplog-reader/pull/6/files#diff-04c6e90faac2675aa89e2176d2eec7d8R7 ?

File: src/index.js (hunk 1 above)
Comment (rick-shar-ww) [reply src/index.js]: Nice!

File: README.md (hunk 3 above)
Comment (will123195) [reply README.md]: Each process has multiple workers, but `workersPerOplog` is respected across processes. I'll add clarification about this.

Comment (will123195) [reply README.md]: But yes, setting to `1` would remove the redundancy for each oplog.

**Comment by will123195** [comment]
fixed

File: README.md (hunk 3 above)
Comment (wwbenomathew) [reply README.md]: Ok, thanks for the clarification

**Review by wwbenomathew — COMMENTED** [review]
Looking solid!

**Pull request merged**